src/components/calcite-inline-editable/calcite-inline-editable.tsx

```diff
@@ -0,0 +1,276 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Listen,
+  Prop,
+  VNode,
+  Watch
+} from "@stencil/core";
+import { getElementProp } from "../../utils/dom";
+import { TEXT } from "./resources";
+
+@Component({
+  tag: "calcite-inline-editable",
+  scoped: true,
+  styleUrl: "calcite-inline-editable.scss"
+})
+export class CalciteInlineEditable {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el!: HTMLCalciteInlineEditableElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Props
+  //
+  //--------------------------------------------------------------------------
+
+  /** specify whether editing can be enabled */
+  @Prop({ reflect: true }) disabled = false;
+
+  @Watch("disabled")
+  disabledWatcher(disabled: boolean): void {
+    this.inputElement.disabled = disabled;
+  }
+
+  /** specify whether the wrapped input element is editable, defaults to false */
+  @Prop({ mutable: true, reflect: true }) editingEnabled = false;
+
+  /** specify whether the confirm button should display a loading state, defaults to false */
+  @Prop({ mutable: true, reflect: true }) loading = false;
+
+  /** specify whether save/cancel controls should be displayed when editingEnabled is true, defaults to false */
+  @Prop({ reflect: true }) hasControls = false;
+
+  /** specify text to be user for the enable editing button's aria-label, defaults to `Click to edit` */
+  @Prop({ reflect: true }) intlEnableEditing = TEXT.intlEnablingEditing;
+
+  /** specify text to be user for the cancel editing button's aria-label, defaults to `Cancel` */
+  @Prop({ reflect: true }) intlCancelEditing = TEXT.intlCancelEditing;
+
+  /** specify text to be user for the confirm changes button's aria-label, defaults to `Save` */
+  @Prop({ reflect: true }) intlConfirmChanges = TEXT.intlConfirmChanges;
+
+  /** specify the scale of the inline-editable component, defaults to the scale of the wrapped calcite-input or the scale of the closest wrapping component with a set scale */
+  @Prop({ reflect: true }) scale?: "s" | "m" | "l";
+
+  /** specify the theme of the inline-editable component, defaults to the theme of the wrapped calcite-input or the theme of the closest wrapping component with a set theme */
+  @Prop({ reflect: true }) theme?: "light" | "dark";
+
+  /** when has-controls, specify a callback to be executed prior to disabling editing. when provided, loading state will be handled automatically. */
+  @Prop() afterConfirm?: () => Promise<any>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Getters/Setters
```

Comment:
I think this could just be moved to `Private methods` right? It seems like it's just a private method